src/txn/txn_timestamp.c

```diff
@@ -107,27 +107,38 @@ __wt_txn_parse_timestamp(WT_SESSION_IMPL *session,
 
 #if WT_TIMESTAMP_SIZE == 8
 	{
-	static const u_char hextable[] = {
-	    0,  0,  0,  0,  0,  0,  0,  0,
-	    0,  0,  0,  0,  0,  0,  0,  0,
-	    0,  0,  0,  0,  0,  0,  0,  0,
-	    0,  0,  0,  0,  0,  0,  0,  0,
-	    0,  0,  0,  0,  0,  0,  0,  0,
-	    0,  0,  0,  0,  0,  0,  0,  0,
-	    0,  1,  2,  3,  4,  5,  6,  7,
-	    8,  9,  0,  0,  0,  0,  0,  0,
-	    0, 10, 11, 12, 13, 14, 15,  0,
-	    0,  0,  0,  0,  0,  0,  0,  0,
-	    0,  0,  0,  0,  0,  0,  0,  0,
-	    0,  0,  0,  0,  0,  0,  0,  0,
-	    0, 10, 11, 12, 13, 14, 15
+	static const char hextable[] = {
+	    -1, -1,  -1,  -1,  -1,  -1,  -1,  -1,
+	    -1, -1,  -1,  -1,  -1,  -1,  -1,  -1,
```

Comment:
@bvpvamsikrishna, we only document lower case hex digits, and the other implementation of the timestamp parser (for timestamp sizes other than 8) only supports lower case.  So this brings the two implementations into sync.

Reply:
Actually, I was looking forward to having support for capital digits, as we could then use some simpler methods to producing these config strings in MongoDB.  That is, the toString() method only outputs capital hexadecimal digits.   It's only a cosmetic improvement, however.

Reply:
The current/original implementation does accept capital hex letters per line 119 in the diff above.

Reply:
True, but the first release of txn_timestamp.c actually used __wt_nhex_to_raw() for all timestamp parsing, and I haven't yet changed the MongoDB code to use capitals to take advantange of the new parsing that has since been introduced.  If support for capital hex goes away, it won't break anything on our side.